flightControl/include/AvState.h

```diff
@@ -1,51 +1,74 @@
-// This is the header file for the AvState class, which is a part of the flightControl module.
-// The AvState class is a finite state machine (FSM) that represents the state of the rocket.
-// The class has a constructor and a destructor, as well as a number of functions that transition from one state to others possible states.
-
-#include <iostream>
-#include <string>
-#include "AvData.h"
-
-#ifndef AVSTATE_H
-#define AVSTATE_H
-
-
-enum class State
-{
-    IDLE,
-    LANDED,
-    DESCENT,
-    ASCENT,
-    CALIBRATION,
-    ERRORGROUND,
-    ERRORFLIGHT,
-    THRUSTSEQUENCE,
-    MANUAL,
-    ARMED,
-    READY
-};
-
-// Path: AV-Firehorn-Rpi/include/flightControl/AvState.h
-// Compare this snippet from AV-Firehorn-Rpi/src/flightControl/FSM.cpp:
-//     // this function allows to get the current state of the FSM
-// functions that transition from one state to others possible states
-class AvState
-{
-public:
-    // constructor
-    AvState();
-    // destructor
-    ~AvState();
-
-    // this function allows to get the current state of the FSM
-    State getCurrentState();
-    
-    void update(AvData data);
-
-
-    State *possibleStates();
-
-    std::string stateToString(State state);
-};
-
-#endif
+// This is the header file for the AvState class, which is a part of the flightControl module.
+// The AvState class is a finite state machine (FSM) that represents the state of the rocket.
+// The class has a constructor and a destructor, as well as a number of functions that transition from one state to others possible states.
+
+#include <iostream>
+#include <list>
+#include "thresholds.h"
+#include <string>
+#include "data.h"
+#include "PacketDefinition.h"
+
+
+
+#ifndef AVSTATE_H
+#define AVSTATE_H
+
+class UPLink {
+public:
+    int id;
+    int data;
+};
+
+enum class State
+{
+    IDLE,
+    LANDED,
+    DESCENT,
+    ASCENT,
+    CALIBRATION,
+    ERRORGROUND,
+    ERRORFLIGHT,
+    THRUSTSEQUENCE,
+    MANUAL,
+    ARMED
+};
+
+
+// Path: AV-Firehorn-Rpi/include/flightControl/AvState.h
+// Compare this snippet from AV-Firehorn-Rpi/src/flightControl/FSM.cpp:
+//     // this function allows to get the current state of the FSM
+// functions that transition from one state to others possible states
+class AvState
+{
+public:
+    // constructor
+    explicit AvState(const Thresholds& thresholds);
+    // destructor
+    ~AvState();
+
+    // this function allows to get the current state of the FSM
+    State getCurrentState();
+
+
+    void update(SensFiltered data, UPLink uplink, bool status);
+    State* possibleStates();
+    std::string stateToString(State state);
```

Comment:
this is likely useless.

Reply:
oui je dois revoir ton archi et revoir update et enlever les 2 autres